metecho/api/jobs.py

```diff
@@ -146,6 +154,106 @@ def _create_branches_on_github(
     return task_branch_name
 
 
+def create_repository(
+    project: Project,
+    *,
+    user: User,
+    dependencies: Iterable[str],
+    template_repo_owner: str = None,
+    template_repo_name: str = None,
+):
+    """
+    Given a local Metecho Project create and bootstrap the corresponding GitHub repository.
+    """
+    project.refresh_from_db()
+
+    try:
+        # Ensure the user is part of the org that owns the project
+        gh_user = gh_given_user(user)
+        user_orgs = [org.login for org in gh_user.organizations()]
+        if project.repo_owner not in user_orgs:
+            raise ValueError(
+                _(
+                    "Either you are not a member of the %(name)s organization "
+                    "or it hasn't installed the Metecho GitHub app"
+                )
+                % {"name": project.repo_owner}
+            )
+
+        # Create team & repository on GitHub
+        org = get_org_for_repo_creation(project.repo_owner)
+        team = org.create_team(f"{project} Team")
+        team.add_or_update_membership(user.username, role="maintainer")
+        for collaborator in project.github_users:
+            team.add_or_update_membership(collaborator["login"], role="member")
+        repo = org.create_repository(project.repo_name, private=False)
+        team.add_repository(repo.full_name, permission="push")
+        project.repo_id = repo.id
+
+        with temporary_dir():
+            # Populate files from the template repository
+            if template_repo_owner and template_repo_name:
+                zipfile = download_extract_github(
+                    gh_user, template_repo_owner, template_repo_name
+                )
+                zipfile.extractall()
+
+            # Bootstrap repository with CumulusCI
+            branch_name = "main"
+            runtime = CliRuntime()
+            context = {
+                "cci_version": cumulusci.__version__,
+                "project_name": project.repo_name,
+                "package_name": project.repo_name,
```

Comment:
@jerivas Does CCI give us a way to use the `project.description` in the initial repo as well?

![Screen Shot 2022-05-17 at 5 13 47 PM](https://user-images.githubusercontent.com/552316/168911289-b3f7f540-9459-41bf-abbb-e60697a7f156.jpg)

Reply:
Hmm, do you mean to set the repo description that appears in the GH sidebar? Or the contents of the README? For the latter, that's something you would do in the template repo. I'm using [this repo](https://github.com/jerivastest/cci-blueprint) as a template and [new repos](https://github.com/jerivastest/p-2294711889) generated from it seem to inherit its readme file

Reply:
I mean the README. I'm not using a template repo at all, but it seems like CCI is populating with some basic info anyway: https://github.com/jerivastest/Metecho-Test-Org

The wording of that default "Add a brief description..." text just made me think maybe that's info we can pass in through CCI. 🤷

Reply:
I couldn't find anything on the CCI docs, but I don't really know what it does 😅 I've added this to the checklist as well

Reply:
From SFDO: templating in the README is not supported